clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -677,7 +679,7 @@ public synchronized void maybeTransitionToErrorState(RuntimeException exception)
                 || exception instanceof UnsupportedVersionException) {
             transitionToFatalError(exception);
         } else if (isTransactional()) {
-            if (canBumpEpoch() && !isCompleting()) {
+            if (!isTransactionV2Enabled && canBumpEpoch() && !isCompleting()) {
```

Comment:
I wonder if we could just put this isTransactionsV2Enabled in the canBumpEpoch check

Reply:
that's a good idea, yep I'll do that!